polln_shift/models/planning.py

```diff
@@ -41,6 +41,39 @@ def subscribe(self):
         return res
 
 
+class TaskTemplate(models.Model):
+    _inherit = "beesdoo.shift.template"
+
+    def name_get(self):
+        res = []
+        for rec in self:
+            name = "%s %s %s (%s-%s)" % (
+                rec.name,
+                rec.planning_id.name,
+                rec.day_nb_id.name,
+                float_to_time(rec.start_time),
+                float_to_time(rec.end_time)
+            )
+            res.append((rec.id, name))
+        return res
+
+    @api.model
+    def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
+        domain = []
+        time_domain = []
+        FIELDS = ['planning_id', 'name', 'day_nb_id']
+        for n in name.split(" "):
+            if re.search(r"^\(\d{2}:\d{2}-\d{2}:\d{2}\)$", n):
```

Comment:
could you provide an example(s) of matched pattern for readability ?
Also, nitpicking, shouldn't you compile the pattern out of the loop?

Reply:
> could you provide an example(s) of matched pattern for readability ? 
  Yes we are looking for the time (generated by the name get) for example 12:30-15h30 

> Also, nitpicking, shouldn't you compile the pattern out of the loop?
 From the perf point of view the number of iteration depends on the number of token in the search term so between 1 and 4. I don't think it really matter.

Reply:
Good to know, thanks!